src/arm/linux/chipset.c build error with clang

src/arm/linux/chipset.c:4019:52: error: '&&' within '||' [-Werror,-Wlogical-op-parentheses]
 4019 |         if (vendor_a == cpuinfo_arm_chipset_vendor_unisoc && vendor_b == cpuinfo_arm_chipset_vendor_spreadtrum ||
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~
src/arm/linux/chipset.c:4019:52: note: place parentheses around the '&&' expression to silence this warning
 4019 |         if (vendor_a == cpuinfo_arm_chipset_vendor_unisoc && vendor_b == cpuinfo_arm_chipset_vendor_spreadtrum ||
      |                                                           ^
      |             (                                                                                                 )
src/arm/linux/chipset.c:4020:56: error: '&&' within '||' [-Werror,-Wlogical-op-parentheses]
 4019 |         if (vendor_a == cpuinfo_arm_chipset_vendor_unisoc && vendor_b == cpuinfo_arm_chipset_vendor_spreadtrum ||
      |                                                                                                                ~~
 4020 |             vendor_a == cpuinfo_arm_chipset_vendor_spreadtrum && vendor_b == cpuinfo_arm_chipset_vendor_unisoc) {
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/arm/linux/chipset.c:4020:56: note: place parentheses around the '&&' expression to silence this warning
 4020 |             vendor_a == cpuinfo_arm_chipset_vendor_spreadtrum && vendor_b == cpuinfo_arm_chipset_vendor_unisoc) {
      |                                                               ^                                               
      |             (                                                                                                 )
2 errors generated.

--- a/src/arm/linux/chipset.c
+++ b/src/arm/linux/chipset.c
@@ -4016,8 +4016,8 @@ static enum cpuinfo_arm_chipset_vendor disambiguate_chipset_vendor(
 	/* Some UNISOC-based platforms reporting conflicting vendor names depending
 	 * on the source. For phones that report both UNISOC and Spreadtrum, treat it
 	 * as UNISOC. */
-	if (vendor_a == cpuinfo_arm_chipset_vendor_unisoc && vendor_b == cpuinfo_arm_chipset_vendor_spreadtrum ||
-	    vendor_a == cpuinfo_arm_chipset_vendor_spreadtrum && vendor_b == cpuinfo_arm_chipset_vendor_unisoc) {
+	if ((vendor_a == cpuinfo_arm_chipset_vendor_unisoc && vendor_b == cpuinfo_arm_chipset_vendor_spreadtrum) ||
+	    (vendor_a == cpuinfo_arm_chipset_vendor_spreadtrum && vendor_b == cpuinfo_arm_chipset_vendor_unisoc)) {
 		return cpuinfo_arm_chipset_vendor_unisoc;
 	}
 

